main.js

```diff
@@ -0,0 +1,386 @@
+let gamesPlayed = 1;
+let startTime = Date.now();
+let turnCount = 0;
+let whoseTurn = "X";
+let occupiedCells = {
+  all: [],
+  X: [],
+  O: []
+};
+let cells = [];
+let startOverButton = document.getElementById("startOver");
+startOverButton.style = "opacity: 0";
+let timer = document.getElementById("timer");
+let playerX = {
+  name: "Player X",
+  type: "Human",
+  winCount: 0
+};
+let playerO = {
+  name: "Player O",
+  type: "Human",
+  winCount: 0
+};
+let statusMessage = document.getElementById("status");
+statusMessage.textContent = XOToPlayerName(whoseTurn) + "'s turn!";
+let changeNamesButton = document.getElementById("changeNames");
+let currentState = "gameStarted";
+let winningTriple = [];
+let playerXNameInput = document.getElementById("playerXName");
+let playerONameInput = document.getElementById("playerOName");
+let playerXWinCount = document.getElementById("playerXwins");
+let playerOWinCount = document.getElementById("playerOwins");
+
+let states = {
+  waiting: { canChangeTo: ["gameStarted"] },
+  gameStarted: { canChangeTo: ["gameFinished"] },
+  gameFinished: { canChangeTo: ["gameStarted"] }
+};
+
+function restartGame() {
+  resetTimer();
+  runTimer();
+  gamesPlayed++;
+  turnCount = 0;
+  if (gamesPlayed % 2 === 0) {
+    whoseTurn = "X";
+  } else {
+    whoseTurn = "O";
+  }
+  occupiedCells = {
+    all: [],
+    X: [],
+    O: []
+  };
+  startOverButton.style = "opacity: 0";
+  statusMessage.textContent = XOToPlayerName(whoseTurn) + "'s turn!";
+  currentState = "gameStarted";
+  winningTriple = [];
+  for (cell of cells) {
+    cell.textContent = "";
+    cell.style.color = "#22ff00";
+  }
+  console.log("restarting " + gamesPlayed + "th game...");
+  computerPlays();
+}
+
+function enterState(newState) {
```

Comment:
Even though (or perhaps because) this is just copy pasted code from past projects, you should try to rename things or at least add some comments to explain what this is doing in terms of this program. This will not only make your code more readable, but also can help you get a deeper understanding of how the function works.